src/build/specialize_types/IR.zig

```diff
@@ -125,129 +147,157 @@ pub const Expr = union(enum) {
     pattern_vars: TypedIdent.Slice,
+    /// expression to the right of `=`
     expr: Expr.Idx,
+    /// type of the expression to the right of `=`
     expr_type: Type.Idx,
-    /// todo
+    /// List of Def
     pub const List = collections.SafeMultiList(@This());
-    /// todo
+    /// Sublist reference into Def.List
     pub const Slice = List.Slice;
 };
 
-/// todo
+/// Branch of a `when` expression, e.g. `Green -> "green"`
 pub const WhenBranch = struct {
     /// The pattern(s) to match the value against
     patterns: Pattern.NonEmptySlice,
     /// A boolean expression that must be true for this branch to be taken
     guard: ?Expr.Idx,
     /// The expression to produce if the pattern matches
     value: Expr.Idx,
-    /// todo
+    /// List of WhenBranch
     pub const List = collections.SafeList(@This());
-    /// todo
+    /// Reference to non-empty sublist into WhenBranch.List
     pub const NonEmptySlice = List.NonEmptySlice;
 };
 
-/// todo
+/// e.g. `|x| x * 2`
 pub const Function = struct {
     args: Pattern.Slice,
     return_type: Type.Idx,
+    /// ??? Is expr the body of the function?
     expr: Expr.Idx,
 };
 
-/// todo
+/// ??? For Record destructuring, e.g. `{ x, y } = coords(point)`
 pub const StructDestruct = struct {
+    /// ???
     ident: Ident.Idx,
+    /// ???
     field: Ident.Idx,
     kind: Kind,
-    /// todo
+    /// ???
     pub const Kind = union(enum) {
+        /// ???
         required,
+        /// ???
         guard: Pattern.Typed,
     };
-    /// todo
+    /// List of StructDestruct
     pub const List = collections.SafeMultiList(@This());
-    /// todo
+    /// Reference to sublist into StructDestruct.List
     pub const Slice = List.Slice;
 };
 
-/// todo
+/// Represents a pattern used in pattern matching e.g. `Ok x` as part of a when branch.
 pub const Pattern = union(enum) {
+    /// e.g. `x`
     identifier: Ident.Idx,
+    /// e.g. `.. as tail` in `[head, .. as tail] -> prepend(head, tail)`
     as: struct {
+        /// e.g. `..` in `.. as tail`
         pattern: Pattern.Idx,
+        /// e.g. `tail` in `.. as tail`
         ident: Ident.Idx,
     },
     str_literal: StringLiteral.Idx,
     number_literal: base.Literal.Num,
+    /// e.g. `Ok(x)` in `Ok(x) ->`
     applied_tag: struct {
+        /// Type of tag union e.g. `Result`
         tag_union_type: Type.Idx,
         tag_name: Ident.Idx,
+        /// e.g. `x` in `Ok(x)`
         args: Pattern.Slice,
     },
+    /// e.g. `{x, y}` in `{x, y} -> Point(x, y)`
     struct_destructure: struct {
         struct_type: Type.Idx,
+        /// ??? e.g. `x` and `y` in `{x, y}`
         destructs: StructDestruct.Slice,
+        /// e.g. `..` in `{x, y, ..}`
         opt_spread: ?Pattern.Typed,
     },
+    /// e.g. [Foo, Bar, ..]
     list: struct {
         elem_type: Type.Idx,
+        /// e.g. Foo and Bar in `[Foo, Bar, ..]`
         patterns: Pattern.Slice,
-
+        /// refers to e.g. `.. as tail` in `[head, .. as tail]`
```

Comment:
Minor syntax change, we are moving to `..tail` instead of `.. as tail`. The `as` example above should use something else, like `{ x, y } as data`.